var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py

```diff
@@ -304,7 +304,7 @@ def write_config_file(self, flags, path, compilers):
             # Tolerate missing compilers.
             # Initially, we installed icx/ifx/icc/ifort into a single prefix.
             # Starting in 2024, there is no icc. 2023.2.3 does not have an ifx.
-            if os.path.exists(compiler):
+            if os.path.exists(path.join(compiler)):
```

Comment:
path.join with 1 argument doesn't do anything

Reply:
Variable path here is a an instance of  `spack.util.prefix.Prefix`. This class has  method `join` which concatenate its argument to a prefix and returns it.

Reply:
I was thinking it was os.path.join. Thanks for the explanation.